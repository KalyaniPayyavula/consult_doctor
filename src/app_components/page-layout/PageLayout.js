import React from "react";
import { MelodyBox } from "../../basic_components";
import PropTypes from "prop-types";

const PageLayout = ({ header, content, footer, dataTestId }) => {
  return (
    <MelodyBox direction="column" data-testid={`${dataTestId}-page-layout`}>
      <MelodyBox border={{ side: "bottom", color: "gray" }}>{header}</MelodyBox>
      <MelodyBox pad="small" background="light-2">
        {content}
      </MelodyBox>
      <MelodyBox background="gray">{footer}</MelodyBox>
    </MelodyBox>
  );
};

PageLayout.propTypes = {
  header: PropTypes.node,
  content: PropTypes.node,
  footer: PropTypes.node,
  dataTestId: PropTypes.string
};

export default PageLayout;
