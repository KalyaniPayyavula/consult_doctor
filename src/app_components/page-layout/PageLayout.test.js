import React from "react";
import { render } from "@testing-library/react";
import PageLayout from "./PageLayout";

test("test snapshot for page layout", () => {
  const header = <p> Header </p>;
  const content = <p> Content </p>;
  const footer = <p> Footer </p>;
  const { container } = render(
    <PageLayout header={header} content={content} footer={footer} />
  );
  expect(container).toMatchSnapshot();
});
