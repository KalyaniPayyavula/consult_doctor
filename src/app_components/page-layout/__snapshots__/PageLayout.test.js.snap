// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot for page layout 1`] = `
<div>
  <div
    class="StyledBox-sc-13pk1d4-0 IRSNj"
    data-testid="undefined-page-layout"
  >
    <div
      class="StyledBox-sc-13pk1d4-0 cwscoI"
    >
      <p>
         Header 
      </p>
    </div>
    <div
      class="StyledBox-sc-13pk1d4-0 ccvWIr"
    >
      <p>
         Content 
      </p>
    </div>
    <div
      class="StyledBox-sc-13pk1d4-0 fZCeFG"
    >
      <p>
         Footer 
      </p>
    </div>
  </div>
</div>
`;
